MantidQt/CustomInterfaces/src/Tomography/TomographyIfacePresenter.cpp

```diff
@@ -554,14 +555,97 @@ void TomographyIfacePresenter::processRunRecon() {
   // center of rotation and regions
   m_model->setImageStackPreParams(m_view->currentROIEtcParams());
 
+  // we have to branch out to handle local and remote somewhere
   try {
-    m_model->doSubmitReconstructionJob(m_view->currentComputeResource());
+    const bool local = isLocalResourceSelected();
+
+    std::string runnable;
+    std::vector<std::string> args;
+    std::string allOpts;
+
+    m_model->prepareSubmissionArguments(local, runnable, args, allOpts);
+    if (local) {
+      setupAndRunLocalReconstruction(runnable, args, allOpts);
+    } else {
+      // get the actual compute resource name
+      const std::string computingResouce = m_view->currentComputeResource();
+      m_model->doRemoteRunReconstructionJob(computingResouce, runnable,
+                                            allOpts);
+    }
   } catch (std::exception &e) {
     m_view->userWarning("Issue when trying to start a job", e.what());
   }
   processRefreshJobs();
 }
 
+void TomographyIfacePresenter::setupAndRunLocalReconstruction(
+    const std::string &runnable, const std::vector<std::string> &args,
+    const std::string &allOpts) {
+
+  if (m_reconRunning) {
+    auto result = m_view->userConfirmation(
+        "Reconstruction is RUNNING",
+        "Are you sure you want to<br>cancel the running reconstruction?");
+
+    if (!result) {
+      // user clicked NO, so we don't terminate the running reconstruction and
+      // simply return
+      return;
+    }
+  }
+
+  // this kills the previous thread forcefully
+  m_workerThread.reset();
+  auto *worker = new MantidQt::CustomInterfaces::TomographyProcess();
+  m_workerThread = Mantid::Kernel::make_unique<TomographyThread>(this, worker);
+
+  // we do this so the thread can independently read the process' output and
+  // only signal the presenter after it's done reading and has something to
+  // share so it doesn't block the presenter
+  connect(m_workerThread.get(), SIGNAL(stdOutReady(QString)), this,
+          SLOT(readWorkerStdOut(QString)));
+  connect(m_workerThread.get(), SIGNAL(stdErrReady(QString)), this,
+          SLOT(readWorkerStdErr(QString)));
+
+  // remove the user confirmation for running recon, if the recon has finished
+  connect(m_workerThread.get(), SIGNAL(workerFinished()), this,
+          SLOT(workerFinished()));
+
+  connect(worker, SIGNAL(started()), this, SLOT(addProcessToJobList()));
+
+  m_model->doLocalRunReconstructionJob(runnable, args, allOpts, *m_workerThread,
+                                       *worker);
+  m_reconRunning = true;
+}
+
+void TomographyIfacePresenter::workerFinished() { m_reconRunning = false; }
```

Comment:
The comment for this method seems to have moved down to `reconProcessFailedToStart`